### PR TITLE
Feature/categories

### PR DESCRIPTION
**Pull request by Avareco**

### CODE REVIEW

File: src/components/Header.vue [hunk 1]
```diff
@@ -1,3 +1,3 @@
 <template>
-  <header class="h-4rem">Header</header>
+  <header class="h-4rem bg-white">Header</header>
```
Comment (hollote) [inline src/components/Header.vue:2]: what about themes support?

File: src/components/MenuLeft.vue [hunk 2]
```diff
@@ -1,3 +1,34 @@
...
+  <nav
+    class="left-menu bg-white"
+    :class="nodes.length > 14 ? 'overflow-y-scroll' : 'overflow-hidden'"
+  >
+    <PanelMenu class="mt-3" :model="nodes" />
+  </nav>
 </template>
+
+<style>
+.p-panelmenu .p-panelmenu-header .p-panelmenu-header-content {
```
Comment (hollote) [inline src/components/MenuLeft.vue:17]: can you use existing classes instead of overriding them in CSS?

File: src/views/ItemPage.vue [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    (breadcrumb) =>
+      rootSectionKey(breadcrumb.key) === rootSectionKey(currentItem.value.key)
+  );
+}
+function rootSectionKey(nodeKey: string): string {
+  if (!nodeKey.includes("_")) return nodeKey;
+  return nodeKey.slice(0, nodeKey.indexOf("_"));
+}
+
+function searchItem(element: IItem): IItem | null {
```
Comment (hollote) [inline src/views/ItemPage.vue:32]: probably this can be re-written

**Review by hollote — CHANGES_REQUESTED** [review]
rewrite breadcrumbs stuff, possibly do cleanup in CSS

File: src/components/Header.vue (hunk 1 above)
Comment (Avareco) [reply src/components/Header.vue:2]: theme support added

File: src/components/MainPage.vue [hunk 4]
```diff
@@ -1,3 +1,11 @@
 <template>
-  <main class="h-full">Main</main>
+  <main class="main bg-white p-3 mt-4">
+    <RouterView />
+  </main>
 </template>
+
+<style scoped>
+.main {
```
[the remark this answers is not in the document]
Comment (Avareco) [reply src/components/MainPage.vue:8]: Yepp, fixed

File: src/components/MenuLeft.vue (hunk 2 above)
Comment (Avareco) [reply src/components/MenuLeft.vue:17]: Ur right, I can, fixed

File: src/views/ItemPage.vue (hunk 3 above)
Comment (Avareco) [reply src/views/ItemPage.vue:32]: fixed

File: src/components/MenuLeft.vue [hunk 5]
```diff
@@ -1,3 +1,22 @@
...
+    :class="nodes.length > 14 ? 'overflow-y-scroll' : 'overflow-hidden'"
+  >
+    <PanelMenu class="border-0" :model="nodes" />
+  </nav>
 </template>
+
+<style>
+.left-menu {
+  height: 100%;
+  min-width: 400px;
```
Comment (hollote) [inline src/components/MenuLeft.vue:19]: please do not use pixels

File: src/views/ItemPage.vue [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+<style>
+.p-menuitem-link {
+  box-shadow: none !important;
+}
+.p-inputnumber-input {
+  width: 50px;
+}
+.close-button {
+  position: absolute;
+  top: -12px;
```
Comment (hollote) [inline src/views/ItemPage.vue:133]: replace pixels

**Review by hollote — CHANGES_REQUESTED** [review]
Please remove pixels

File: src/components/MenuLeft.vue (hunk 5 above)
Comment (Avareco) [reply src/components/MenuLeft.vue]: fixed

File: src/views/ItemPage.vue (hunk 6 above)
Comment (Avareco) [reply src/views/ItemPage.vue]: fixed

File: src/views/ItemPage.vue [hunk 7]
```diff
@@ -121,23 +122,21 @@ watch(
   </div>
 </template>
 
-<style>
+<style scoped>
```
Comment (Avareco) [inline src/views/ItemPage.vue:125]: @hollote, I added scoped in styles, but some styles doesn't work(.p-inputtext, .p-menuitem-link, .close-button > span). What should I do?

Comment (hollote) [reply src/views/ItemPage.vue:125]: check `::deep` for vue styles

Comment (Avareco) [reply src/views/ItemPage.vue:125]: done

File: src/views/ItemPage.vue [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+    return item[arrIndex];
+  } else {
+    return getCurrentEvent(item[arrIndex].items, path);
+  }
+}
+
+function createBreadcrumbs(item: IItem, path: string[], breadcrumbs) {
+  const arrIndex = Number(path.shift()) - 1;
+  breadcrumbs = [...breadcrumbs, item[arrIndex]];
+  if (path.length === 0) {
```
Comment (hollote) [inline src/views/ItemPage.vue:26]: same here

**Review by hollote — APPROVED** [review]

File: src/views/ItemPage.vue (hunk 8 above)
Comment (Avareco) [reply src/views/ItemPage.vue:26]: will be fixed in next pull request

File: src/views/ItemPage.vue [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+const route = useRoute();
+const store = useTreeNodes();
+const breadcrumbs = ref<IItem>([]);
+
+const items = store.$state;
+const currentItem = ref<IItem>({});
+
+function getCurrentEvent(item: IItem, path: string[]): IItem {
+  const arrIndex = Number(path.shift()) - 1;
+  if (path.length === 0) {
```
[the remark this answers is not in the document]
Comment (Avareco) [reply src/views/ItemPage.vue:16]: will be fixed in next pull request

**Pull request merged**